examples/j1j2_su/main.jl

```diff
@@ -0,0 +1,121 @@
+using Markdown #hide
+md"""
+# Three-site simple update for the $J_1$-$J_2$ model
+
+In this example, we will use [`SimpleUpdate`](@ref) imaginary time evolution to treat
+the two-dimensional $J_1$-$J_2$ model, which contains next-nearest neighbor interactions:
+
+```math
+H = J_1 \sum_{\langle i,j \rangle} \mathbf{S}_i \cdot \mathbf{S}_j
++ J_2 \sum_{\langle \langle i,j \rangle \rangle} \mathbf{S}_i \cdot \mathbf{S}_j
+```
+
+Note that the $J_1$-$J_2$ model exhibits a $U(1)$ spin rotation symmetry, which we want to 
+exploit here. The goal will be to calculate the energy at $J_1 = 1$ and $J_2 = 1/2$, first
+using the simple update algorithm and then, to refine the energy estimate, using AD-based
+variational PEPS optimization.
+
+We first import all required modules and seed the RNG:
+"""
+
+using Random
+using TensorKit, PEPSKit
+Random.seed!(2025);
+
+md"""
+## Simple updating a challenging phase
+
+Let's start by initializing an `InfiniteWeightPEPS` for which we set the required
+parameters as well as physical and virtual vector spaces. Since the $J_1$-$J_2$ model has
+*next*-neighbor interactions, the simple update algorithm requires a $2 \times 2$ unit cell:
+"""
+
+Dbond, χenv, symm = 4, 32, U1Irrep
+trscheme_env = truncerr(1e-10) & truncdim(χenv)
+Nr, Nc, J1 = 2, 2, 1.0
+
+## random initialization of 2x2 iPEPS with weights and CTMRGEnv (using real numbers)
+Pspace = Vect[U1Irrep](1//2 => 1, -1//2 => 1)
+Vspace = Vect[U1Irrep](0 => 2, 1//2 => 1, -1//2 => 1)
+Espace = Vect[U1Irrep](0 => χenv ÷ 2, 1//2 => χenv ÷ 4, -1//2 => χenv ÷ 4)
+wpeps = InfiniteWeightPEPS(rand, Float64, Pspace, Vspace; unitcell=(Nr, Nc));
+
+md"""
+The value $J_2 / J_1 = 0.5$ is close to a possible spin liquid phase, which is challenging
+for SU to produce a relatively good state from random initialization. Therefore, we shall
+gradually increase $J_2 / J_1$ from 0.1 to 0.5, each time initializing on the previously
+evolved PEPS:
+"""
+
+dt, tol, maxiter = 1e-2, 1e-8, 30000
+check_interval = 4000
+trscheme_peps = truncerr(1e-10) & truncdim(Dbond)
+alg = SimpleUpdate(dt, tol, maxiter, trscheme_peps)
+for J2 in 0.1:0.1:0.5
+    H = real( ## convert Hamiltonian `LocalOperator` to real floats
+        j1_j2_model(ComplexF64, symm, InfiniteSquare(Nr, Nc); J1, J2, sublattice=false),
+    )
```

Comment:
```suggestion
    H = j1_j2_model(Float64, symm, InfiniteSquare(Nr, Nc); J1, J2, sublattice=false)
```
I know this used to fail since we tried to construct `Sy`, was this not fixed in the meantime?

Reply:
No, I still cannot directly set `T = Float64` because of `S_y` in `MPSKitModels` v0.4.2.